far/global.hpp

```diff
@@ -42,14 +42,58 @@ THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #include "platform.hpp"
 
 // Common:
+#include "common/bytes_view.hpp"
 #include "common/preprocessor.hpp"
 
 // External:
 
 //----------------------------------------------------------------------------
 
-enum class lng;
-enum class search_case_fold;
+struct global_search_replace_params
+{
+	NONCOPYABLE(global_search_replace_params);
+	NONMOVABLE(global_search_replace_params);
```

Comment:
Ah, I thought of these things differently. More like a high-level design choice, which may also reflect performance considerations. Something on the lines of, "this thing, while technically copyable (or movable), do not need to be copied (or moved) around by design. To prevent accidental expensive operations, let's outright ban them. If you really think you need a copy, change the code here, but you better have a really good reason."

Anyway, I will remove `NONMOVABLE`.